pypose/module/pnp.py

```diff
@@ -0,0 +1,483 @@
+import torch
+from functools import partial
+from functorch import vmap
+from pypose.optim import GaussNewton
+
+
+class EfficientPnP(torch.nn.Module):
+    """
+        EPnP Solver - a non-iterative O(n) solution to the PnP problem.
+        as described in:
+
+        Francesc Moreno-Noguer, Vincent Lepetit, Pascal Fua.
+        Accurate Non-Iterative O(n) Solution to the PnP Problem.
+        In Proceedings of ICCV, 2007.
+        source: https://github.com/cvlab-epfl/EPnP
+
+        Args:
+            gauss_newton (bool): Use Gauss-Newton to refine the solution.
+            naive_ctrl_pts (bool): Use naive control points selection method.
+
+        Note:
+            The Gauss-Newton optimization step isn't exactly consistent with the implementation in the original paper's
+            implementation, but it relies on the same optimization objective function.
+    """
+    six_indices_pair = torch.tensor([(0, 1), (0, 2), (0, 3), (1, 2), (1, 3), (2, 3)], device='cpu').T
+    six_indices = [(0 * 4 + 1), (0 * 4 + 2), (0 * 4 + 3), (1 * 4 + 2), (1 * 4 + 3), (2 * 4 + 3)]
+    ten_indices_pair = torch.tensor([(0, 0),
+                                     (0, 1), (1, 1),
+                                     (0, 2), (1, 2), (2, 2),
+                                     (0, 3), (1, 3), (2, 3), (3, 3)], device='cpu').T
+    # equal mask [ True, False,  True, False, False,  True, False, False, False,  True]
+    multiply_mask = torch.tensor([1., 2., 1., 2., 2., 1., 2., 2., 2., 1.], device='cpu')
```

Comment:
move to `init()` and define as `register_buffer()` so that it can follow the device of the module.